mindmeld/system_entity_recognizer.py

```diff
@@ -41,61 +50,194 @@ class SystemEntityError(Exception):
+            else:
+                SystemEntityRecognizer._instance = (
+                    DummySystemEntityRecognizer.get_instance()
+                )
+        return SystemEntityRecognizer._instance
+
+    @abstractmethod
+    def parse(self, sentence, **kwargs):
+        """Calls System Entity Recognizer service API to extract numerical entities from a sentence.
+
+        Args:
+            sentence (str): A raw sentence.
+
+        Returns:
+            (tuple): A tuple containing:
+                - response (list, dict): Response from the System Entity Recognizer service that \
+                consists of a list of dicts, each corresponding to a single prediction or just a \
+                dict, corresponding to a single prediction.
+                - response_code (int): http status code.
+        """
+        pass
+
+    @abstractmethod
+    def resolve_system_entity(self, query, entity_type, span):
+        """Resolves a system entity in the provided query at the specified span.
+
+        Args:
+            query (Query): The query containing the entity
+            entity_type (str): The type of the entity
+            span (Span): The character span of the entity in the query
+
+        Returns:
+            Entity: The resolved entity
+
+        Raises:
+            SystemEntityResolutionError
+        """
+        pass
+
+    @abstractmethod
+    def get_candidates(self, query, entity_types=None, **kwargs):
+        """Identifies candidate system entities in the given query.
+
+        Args:
+            query (Query): The query to examine
+            entity_types (list of str): The entity types to consider
+
+        Returns:
+            list of QueryEntity: The system entities found in the query
+        """
+        pass
+
+    @abstractmethod
+    def get_candidates_for_text(self, text, entity_types=None, **kwargs):
+        """Identifies candidate system entities in the given text.
+
+        Args:
+            text (str): The text to examine
+            entity_types (list of str): The entity types to consider
+
+        Returns:
+            list of dict: The system entities found in the text
+        """
+        pass
+
+
+class DummySystemEntityRecognizer(SystemEntityRecognizer):
+    """
+    This is a dummy recognizer which returns empty list and NO_RESPONSE_CODE.
     """
 
     _instance = None
 
-    def __init__(self, app_path=None):
+    def __init__(self):
+        if not DummySystemEntityRecognizer._instance:
+            DummySystemEntityRecognizer._instance = self
+        else:
+            raise SystemEntityError("DummySystemEntityRecognizer is a singleton")
+
+    @staticmethod
+    def get_instance():
+        if not DummySystemEntityRecognizer._instance:
+            DummySystemEntityRecognizer()
+
+        return DummySystemEntityRecognizer._instance
+
+    def parse(self, sentence, **kwargs):
+        return [], NO_RESPONSE_CODE
```

Comment:
I think the response code should be 200 here.

Reply:
I dont think `NO_RESPONSE_CODE ` is needed.

Reply:
This is following the existing behavior. We can update this behavior to return 200 instead of -1, but we do need to return the Response Code

Reply:
Sure, I am saying to return 200.

Reply:
Done